File: javascript/bob/bob.js
export const hey = message => {
  const askYell = /^(?=.*[A-Z])[^a-z]+\?+$/;
  const ask = /\?\s*$/;
  const yell = /^(?=.*[A-Z])[^a-z]+$/;
  const silence = /^\s*$/;

  if (askYell.test(message)) return 'Calm down, I know what I\'m doing!';
  if (ask.test(message)) return 'Sure.';
  if (yell.test(message)) return 'Whoa, chill out!';
  if (silence.test(message)) return 'Fine. Be that way!';

  return 'Whatever.';
};

File: kotlin/bob/src/main/kotlin/Bob.kt
object Bob {
    fun hey(input: String): String {
        val i = input.trim()

        val isShouting = i.any(Char::isLetter) && input == i.uppercase()
        val isAsking = i.endsWith("?")
        val isSilence = i.isBlank()

        if (isShouting && isAsking) return "Calm down, I know what I'm doing!"
        if (isShouting) return "Whoa, chill out!"
        if (isAsking) return "Sure."
        if (isSilence) return "Fine. Be that way!"
        return "Whatever."
    }
}
